src/OidcClientSettings.ts

```diff
@@ -92,6 +92,8 @@ export interface OidcClientSettings {
      * Otherwise, they are added to an array as distinct objects for the claim type. (default: false)
      */
     mergeClaims?: boolean;
+    /** Indicates if the library should use the legacy merge behavior, which mutates string claims into arrays whenever the data is updated on the remote. (default: false) */
+    mergeClaimsLegacyBehavior?: boolean;
```

Comment:
funny: we introduce an option and make it directly deprecated, but otherwise we can not get rid of it any time soon...

Reply:
@Badisi Do you see any use case for the broken old legacy merging?

Reply:
This whole "merge claims mess" was introduced back in 2019 with this PR [#754](https://github.com/IdentityModel/oidc-client-js/pull/754) but existed before that.
Complains followed [(#861)](https://github.com/IdentityModel/oidc-client-js/issues/861) after the PR and the `mergeClaims: boolean` option was introduced to mitigate that.

I have no idea why this behavior of keeping the old and new values into arrays was made, and I don't know any use cases that would required that.. Maybe @brockallen could enlighten us here ? But my understanding of `mergeClaims()` is that it is used to merge the `IdToken` with the response of `UserInfo`. So I could see why it was necessary somehow to keep values of keys that would exists in the two but were different.

My concerns:

1. With @marcoreni new proposal, we are making `UserInfo` the master, as we are overriding any existing key of `claims1` with the value of `claims2`. Is it a good assumption ? You tell me.

2. Also, AFAIU, with this new algorithm we are keeping any keys in `claims1` that do not exist in `claims2` and replacing the existing ones or merging them if there are objects. But for arrays, the logic is different as we are just overriding them, where I would have expected to also keep the existing values and only add new ones or replace existing).
   ``` 
   claims1 { a: 'a', b: [1,2,3], c: { d: 'd' } }
   claims2 { b: [4], c: { e: 'e' } } 
   actual results: { a: 'a', b: [4], c: { d: 'd', e: 'e' } }
   my expectation: { a: 'a', b: [1,2,3,4], c: { d: 'd', e: 'e' } }
   ```

3. Even with `mergeClaims: false` we are still, somehow, merging the two as `claims1` would receive any new keys from `claims2` and got its already existing keys overridden. So as a user perspective it doesn't feel quite right.

Reply:
While I'm thinking about it, why are we trying to make this `mergeClaims()` so complex ?

This isn't clear in the RFC, but can we say that the `IdToken` is always a subset of what is returned by `UserInfo` ? And if that's true, then `UserInfo` is always the up-to-date reference and a simple merge like `{ ...claims1, ..claims2 }` should be enough ?

Reply:
@Badisi this PR is part of a larger PR (#877) that aims at adding the capability to refresh UserInfo without refreshing the token.

So, even if UserInfo should be a superset of IdToken (I was also trying to find out some spec about this), if we add the refresh of UserInfo, in order to have a clean and definitive solution we may need to address some more complex scenarios (as I mentioned [here](https://github.com/authts/oidc-client-ts/issues/846#issuecomment-1406430179))

My "ideal solution", regarding UserInfo, is that we should have a "core object", that does NOT contain any of the UserInfo data, that is the master of the information. 
If the client has "loadUserInfo" enabled, the userinfo stuff is retrieved and cached separately, and merged "on demand". Whenever the "userinfo" is updated, the userinfo cache is completely overwritten (so, no leftovers).

This, however, would introduce some heavy refactoring on the library, unless we find some workaroung (ie. not decoding the jwt while validating the response and caching only userinfo inside "profile", then merging on retrieval, something like that).

Reply:
@Badisi Thanks for the excellent summary!

> But my understanding of `mergeClaims()` is that it is used to merge the `IdToken` with the response of `UserInfo`.

I have the same understanding.

To your concern 1:
It can still be controlled via the settings `loadUserInfo`, which is `false` by default. So no surprise here.
To your concern 2:
I agree with you about the arrays logic, lets combine them when merging. There are anyway not many array fields, most of the claims have string or object as type. This is one i have found, which can be an array: `aud?: string | string[];`
To your concern 3:
I agree with you, the existing behavior should be kept if `mergeClaims` is `false`.

Reply:
hmm, when i look at the current test code, it does exactly what @Badisi and myself specified above. Do i miss something?

Reply:
new:
```
...
else if (result[claim] !== value) {
  if (this._settings.mergeClaims) {
      if (typeof value === "object") {
          result[claim] = this._mergeClaims(previousValue as UserProfile, value);
      } else {
          result[claim] = value;
      }
  }
  else {
      result[claim] = [previousValue, value];
  }
}
...
```